tests/ha/control_pod/test_control_pod_softfailure.py

```diff
@@ -0,0 +1,273 @@
+    @pytest.mark.tags("TEST-45498")
+    def test_45498(self):
+        """
+        Verify IAM user creation/deletion and IOs on already created IAM user while performing
+        control pod soft-failure in loop one by one for N control pods.
+        """
+        LOGGER.info("STARTED: Verify IAM user creation/deletion & IOs on already created IAM user "
+                    "while performing control pod soft-failure in loop 1 by 1 for N control pods.")
+
+        LOGGER.info("Step 1: Perform IOs before soft-failure.")
+        self.test_prefix = 'test-45498'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(self.s3_clean.values())[0],
+                                                    log_prefix=self.test_prefix, skipcleanup=True,
+                                                    nsamples=2, nclients=2)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 1: Performed IOs before soft-failure.")
+
+        LOGGER.info("Scale replicas for control pod to %s", self.repl_num)
+        resp = self.node_master_list[0].create_pod_replicas(num_replica=self.repl_num,
+                                                            deploy=self.deply_name)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Total number of control pods in the cluster are: %s", self.repl_num)
+
+        iam_output = Queue()
+        io_output = Queue()
+        event = threading.Event()
+
+        LOGGER.info("Step 2: Starting independent background threads for IOs on already created "
+                    "user and IAM user creation/deletion.")
+        LOGGER.info("Step 2.1: Start IAM user creation/deletion in background during control pod "
+                    "soft-failure.")
+
+        args = {'user_crud': True, 'num_users': self.num_users,
+                'output': iam_output, 'del_users_dict': self.del_users, 'header': self.header}
+        thr_iam = threading.Thread(target=self.ha_obj.iam_bucket_cruds,
+                                   args=(event, ), kwargs=args)
+        thr_iam.daemon = True  # Daemonize thread
+        self.test_io_prefix = 'test-45498-io'
+        args = {'s3userinfo': list(self.s3_clean.values())[0], 'log_prefix': self.test_io_prefix,
+                'nclients': 1, 'nsamples': 8, 'output': io_output, 'setup_s3bench': False}
+        LOGGER.info("Step 2.2: Start IOs in background during control pod soft-failure.")
+        io_thread = threading.Thread(target=self.ha_obj.event_s3_operation, args=(event,),
+                                     kwargs=args)
+        io_thread.daemon = True  # Daemonize thread
+        io_thread.start()
+        thr_iam.start()
+        LOGGER.info("Step 2: Started independent background threads for IOs on already created "
+                    "user and IAM user creation/deletion.")
+
+        LOGGER.info("Step 3: Perform soft-failure of %s control pods in loop and check cluster "
+                    "has failures", self.repl_num - 1)
```

Comment:
test says for N control pods and u r taking N-1?

Reply:
Mistake in log, will correct it.
I am taking a list of control pod only.
`        LOGGER.info("Step 3: Perform soft-failure of %s control pods in loop and check cluster "
                    "has failures", self.repl_num - 1)
        LOGGER.info("Get the list of control pods")
        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.CONTROL_POD_NAME_PREFIX)`